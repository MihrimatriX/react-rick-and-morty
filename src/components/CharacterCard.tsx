import type { Character } from "../types/character";
import { Link } from "react-router-dom";
import { FaUser, FaGlobe } from "react-icons/fa";

interface CharacterCardProps
	extends Pick<Character, "id" | "img" | "name" | "species" | "status"> {}

const statusColors: Record<string, string> = {
	Alive: "bg-green-500",
	Dead: "bg-red-500",
	unknown: "bg-gray-400",
};

const speciesColors: Record<string, string> = {
	Human: "bg-blue-500",
	Alien: "bg-purple-500",
	unknown: "bg-gray-400",
};

const CharacterCard = ({
	id,
	img,
	name,
	species,
	status,
}: CharacterCardProps) => {
	return (
		<Link
			to={`/char/${id}`}
			tabIndex={0}
			aria-label={`${name} detayÄ±na git`}
			className="block focus:outline-none focus:ring-2 focus:ring-cyan-400"
		>
			<div className="w-64 flex flex-col text-left m-2 p-0 rounded-2xl bg-card-light dark:bg-card-dark shadow-xl border-2 border-transparent hover:border-accent-light dark:hover:border-accent-dark transition-all duration-300 hover:scale-105 hover:shadow-2xl backdrop-blur-md group cursor-pointer overflow-hidden relative">
				<div className="relative w-full aspect-square overflow-hidden">
					<img
						src={img}
						alt={name}
						className="w-full h-full object-cover transition-transform duration-500 group-hover:scale-110 group-hover:blur-[1px] rounded-t-2xl"
						loading="lazy"
					/>
					<div className="absolute inset-0 bg-gradient-to-t from-black/60 via-transparent to-transparent pointer-events-none transition-opacity duration-500 group-hover:opacity-80" />
					<span
						className={`absolute top-2 left-2 px-2 py-1 text-xs rounded-full text-white font-bold shadow ${status === "Alive" ? "bg-rick" : status === "Dead" ? "bg-evil" : "bg-portal"}`}
					>
						{status}
					</span>
					<span
						className={`absolute top-2 right-2 px-2 py-1 text-xs rounded-full text-white font-bold shadow ${species === "Human" ? "bg-morty" : species === "Alien" ? "bg-evil" : "bg-portal"}`}
					>
						{species}
					</span>
				</div>
				<main className="flex-1 flex flex-col justify-end bg-white/80 dark:bg-bg-dark/80 p-4 rounded-b-2xl transition-colors duration-300 group-hover:bg-rick/20 dark:group-hover:bg-morty/20">
					<h3 className="font-bold text-lg mb-1 truncate flex items-center gap-2 text-slate-900 dark:text-white">
						<FaUser className="text-rick" /> {name}
					</h3>
					<div className="flex items-center gap-2 text-sm text-slate-700 dark:text-slate-300">
						<FaGlobe className="text-morty" />
						<span>{species}</span>
					</div>
				</main>
			</div>
		</Link>
	);
};

export default CharacterCard;
