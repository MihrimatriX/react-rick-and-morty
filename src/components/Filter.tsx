import type { FilterData } from "../types/filter";
import {
	FaSearch,
	FaFilter,
	FaSyncAlt,
	FaVenusMars,
	FaHeartbeat,
} from "react-icons/fa";
import { Button } from "./ui/Button";

interface FilterProps {
	handleFilter: (filterData: FilterData) => void;
	onResetFilters: () => void;
}

const statusOptions = [
	{ value: "all", label: "All" },
	{ value: "Alive", label: "Alive" },
	{ value: "Dead", label: "Dead" },
	{ value: "unknown", label: "Unknown" },
];
const genderOptions = [
	{ value: "all", label: "All" },
	{ value: "Male", label: "Male" },
	{ value: "Female", label: "Female" },
	{ value: "Genderless", label: "Genderless" },
	{ value: "unknown", label: "Unknown" },
];

const Filter = ({ handleFilter, onResetFilters }: FilterProps) => (
	<section className="w-full flex justify-center mt-8 mb-6 animate-slideDownAndFade">
		<form
			className="flex flex-col lg:flex-row items-center gap-6 bg-card-light/80 dark:bg-card-dark/80 backdrop-blur-md rounded-3xl shadow-2xl border-2 border-accent-light/30 dark:border-accent-dark/30 px-8 py-6 max-w-4xl mx-auto transition-all duration-300 hover:shadow-3xl hover:border-accent-light/50 dark:hover:border-accent-dark/50"
			onSubmit={(e) => e.preventDefault()}
			aria-label="Karakter Filtreleme"
		>
			<div className="flex flex-col gap-2 w-full lg:w-auto">
				<label
					htmlFor="searchBox"
					className="flex items-center gap-2 text-accent-light dark:text-accent-dark font-bold text-lg transition-colors duration-200 hover:text-rick dark:hover:text-morty"
				>
					<FaSearch className="text-xl" /> İsim
				</label>
				<input
					id="searchBox"
					type="text"
					placeholder="Karakter ara..."
					className="px-5 py-3 rounded-xl border-2 border-accent-light/50 dark:border-accent-dark/50 focus:border-rick dark:focus:border-morty focus:ring-4 focus:ring-rick/20 dark:focus:ring-morty/20 bg-white/90 dark:bg-bg-dark/90 text-slate-900 dark:text-white shadow-lg transition-all duration-300 w-full lg:w-64 hover:border-accent-light dark:hover:border-accent-dark hover:shadow-xl hover:scale-[1.02] focus:scale-[1.02]"
					onChange={(e) =>
						handleFilter({
							key: "searchBox",
							value: e.target.value,
						})
					}
					aria-label="Karakter ismi ara"
				/>
			</div>
			<div className="flex flex-col gap-2 w-full lg:w-auto">
				<label
					htmlFor="status"
					className="flex items-center gap-2 text-accent-light dark:text-accent-dark font-bold text-lg transition-colors duration-200 hover:text-rick dark:hover:text-morty"
				>
					<FaHeartbeat className="text-xl" /> Status
				</label>
				<select
					id="status"
					className="px-5 py-3 rounded-xl border-2 border-accent-light/50 dark:border-accent-dark/50 focus:border-rick dark:focus:border-morty focus:ring-4 focus:ring-rick/20 dark:focus:ring-morty/20 bg-white/90 dark:bg-bg-dark/90 text-slate-900 dark:text-white shadow-lg transition-all duration-300 w-full lg:w-40 hover:border-accent-light dark:hover:border-accent-dark hover:shadow-xl hover:scale-[1.02] focus:scale-[1.02] cursor-pointer"
					onChange={(e) =>
						handleFilter({ key: "status", value: e.target.value })
					}
					aria-label="Status seç"
				>
					{statusOptions.map((opt) => (
						<option
							key={opt.value}
							value={opt.value}
							className="bg-white dark:bg-bg-dark text-slate-900 dark:text-white"
						>
							{opt.label}
						</option>
					))}
				</select>
			</div>
			<div className="flex flex-col gap-2 w-full lg:w-auto">
				<label
					htmlFor="gender"
					className="flex items-center gap-2 text-accent-light dark:text-accent-dark font-bold text-lg transition-colors duration-200 hover:text-rick dark:hover:text-morty"
				>
					<FaVenusMars className="text-xl" /> Gender
				</label>
				<select
					id="gender"
					className="px-5 py-3 rounded-xl border-2 border-accent-light/50 dark:border-accent-dark/50 focus:border-rick dark:focus:border-morty focus:ring-4 focus:ring-rick/20 dark:focus:ring-morty/20 bg-white/90 dark:bg-bg-dark/90 text-slate-900 dark:text-white shadow-lg transition-all duration-300 w-full lg:w-40 hover:border-accent-light dark:hover:border-accent-dark hover:shadow-xl hover:scale-[1.02] focus:scale-[1.02] cursor-pointer"
					onChange={(e) =>
						handleFilter({ key: "gender", value: e.target.value })
					}
					aria-label="Gender seç"
				>
					{genderOptions.map((opt) => (
						<option
							key={opt.value}
							value={opt.value}
							className="bg-white dark:bg-bg-dark text-slate-900 dark:text-white"
						>
							{opt.label}
						</option>
					))}
				</select>
			</div>
			<Button
				type="button"
				className="mt-4 lg:mt-8 px-10 py-4 rounded-2xl bg-gradient-to-r from-rick to-morty dark:from-morty dark:to-rick text-white font-bold text-xl shadow-2xl hover:scale-110 hover:shadow-3xl focus:outline-none focus:ring-4 focus:ring-accent-light/50 dark:focus:ring-accent-dark/50 transition-all duration-300 relative overflow-hidden group"
				onClick={onResetFilters}
				aria-label="Filtreleri temizle"
			>
				<FaSyncAlt className="inline-block mr-3 text-2xl group-hover:animate-spin transition-transform duration-300" />
				Temizle
			</Button>
		</form>
	</section>
);

export default Filter;
