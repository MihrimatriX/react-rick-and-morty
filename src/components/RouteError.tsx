import { Link } from "react-router-dom";
import noResultsImg from "../assets/no_results_img.png";

const RouteError = () => (
	<div className="flex flex-col items-center justify-center min-h-[60vh] px-4 text-center animate-fadeIn">
		<img
			src={noResultsImg}
			alt="404 Portal"
			className="w-48 h-48 mb-6 animate-wiggle drop-shadow-lg select-none"
			aria-hidden="true"
		/>
		<h1 className="text-5xl font-extrabold text-cyan-600 dark:text-cyan-400 mb-2">
			404
		</h1>
		<h2 className="text-2xl font-bold mb-2 text-gray-800 dark:text-gray-100">
			You're Lost in Another Universe!
		</h2>
		<p className="mb-6 text-gray-600 dark:text-gray-300 max-w-md">
			The page you're looking for doesn't exist in this universe. Use the portal to return to the home page!
		</p>
		<Link
			to="/"
			className="px-6 py-3 rounded-lg bg-cyan-600 text-white font-semibold shadow hover:bg-cyan-700 focus:outline-none focus:ring-2 focus:ring-cyan-400 transition text-lg animate-fadeIn"
			tabIndex={0}
			aria-label="Return to home page"
			onKeyDown={(e) => {
				if (e.key === "Enter" || e.key === " ")
					(e.target as HTMLElement).click();
			}}
		>
			Return to Home Page
		</Link>
	</div>
);

export default RouteError;
