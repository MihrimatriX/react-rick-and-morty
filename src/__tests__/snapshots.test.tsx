import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import CharacterCard from '../components/CharacterCard';
import Filter from '../components/Filter';
import RouteError from '../components/RouteError';
import { BrowserRouter } from 'react-router-dom';

describe('Snapshot Tests', () => {
	test('CharacterCard matches snapshot', () => {
		const { container } = render(
			<MemoryRouter>
				<CharacterCard
					id={1}
					img="rick.jpg"
					name="Rick Sanchez"
					species="Human"
					status="Alive"
				/>
			</MemoryRouter>
		);
		expect(container).toMatchSnapshot();
	});

	test('Filter matches snapshot', () => {
		const { container } = render(
			<Filter handleFilter={jest.fn()} onResetFilters={jest.fn()} />
		);
		expect(container).toMatchSnapshot();
	});

	test('RouteError matches snapshot', () => {
		const { container } = render(
			<BrowserRouter>
				<RouteError />
			</BrowserRouter>
		);
		expect(container).toMatchSnapshot();
	});
}); 