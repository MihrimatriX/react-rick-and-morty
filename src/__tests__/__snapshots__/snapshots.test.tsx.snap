// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Snapshot Tests CharacterCard matches snapshot 1`] = `
<div>
  <a
    aria-label="Rick Sanchez detayÄ±na git"
    class="block focus:outline-none focus:ring-2 focus:ring-cyan-400"
    data-discover="true"
    href="/char/1"
    tabindex="0"
  >
    <div
      class="w-64 flex flex-col text-left m-2 p-0 rounded-2xl bg-card-light dark:bg-card-dark shadow-xl border-2 border-transparent hover:border-accent-light dark:hover:border-accent-dark transition-all duration-300 hover:scale-105 hover:shadow-2xl backdrop-blur-md group cursor-pointer overflow-hidden relative"
    >
      <div
        class="relative w-full aspect-square overflow-hidden"
      >
        <img
          alt="Rick Sanchez"
          class="w-full h-full object-cover transition-transform duration-500 group-hover:scale-110 group-hover:blur-[1px] rounded-t-2xl"
          loading="lazy"
          src="rick.jpg"
        />
        <div
          class="absolute inset-0 bg-gradient-to-t from-black/60 via-transparent to-transparent pointer-events-none transition-opacity duration-500 group-hover:opacity-80"
        />
        <span
          class="absolute top-2 left-2 px-2 py-1 text-xs rounded-full text-white font-bold shadow bg-rick"
        >
          Alive
        </span>
        <span
          class="absolute top-2 right-2 px-2 py-1 text-xs rounded-full text-white font-bold shadow bg-morty"
        >
          Human
        </span>
      </div>
      <main
        class="flex-1 flex flex-col justify-end bg-white/80 dark:bg-bg-dark/80 p-4 rounded-b-2xl transition-colors duration-300 group-hover:bg-rick/20 dark:group-hover:bg-morty/20"
      >
        <h3
          class="font-bold text-lg mb-1 truncate flex items-center gap-2 text-slate-900 dark:text-white"
        >
          <svg
            class="text-rick"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            />
          </svg>
           
          Rick Sanchez
        </h3>
        <div
          class="flex items-center gap-2 text-sm text-slate-700 dark:text-slate-300"
        >
          <svg
            class="text-morty"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
            />
          </svg>
          <span>
            Human
          </span>
        </div>
      </main>
    </div>
  </a>
</div>
`;

exports[`Snapshot Tests Filter matches snapshot 1`] = `
<div>
  <section
    class="w-full flex justify-center mt-8 mb-6 animate-slideDownAndFade"
  >
    <form
      aria-label="Character Filtering"
      class="flex flex-col lg:flex-row items-center gap-6 bg-card-light/80 dark:bg-card-dark/80 backdrop-blur-md rounded-3xl shadow-2xl border-2 border-accent-light/30 dark:border-accent-dark/30 px-8 py-6 max-w-4xl mx-auto transition-all duration-300 hover:shadow-3xl hover:border-accent-light/50 dark:hover:border-accent-dark/50"
    >
      <div
        class="flex flex-col gap-2 w-full lg:w-auto"
      >
        <label
          class="flex items-center gap-2 text-accent-light dark:text-accent-dark font-bold text-lg transition-colors duration-200 hover:text-rick dark:hover:text-morty"
          for="searchBox"
        >
          <svg
            class="text-xl"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
           Name
        </label>
        <input
          aria-label="Search character name"
          class="px-5 py-3 rounded-xl border-2 border-accent-light/50 dark:border-accent-dark/50 focus:border-rick dark:focus:border-morty focus:ring-4 focus:ring-rick/20 dark:focus:ring-morty/20 bg-white/90 dark:bg-bg-dark/90 text-slate-900 dark:text-white shadow-lg transition-all duration-300 w-full lg:w-64 hover:border-accent-light dark:hover:border-accent-dark hover:shadow-xl hover:scale-[1.02] focus:scale-[1.02]"
          id="searchBox"
          placeholder="Search character..."
          type="text"
        />
      </div>
      <div
        class="flex flex-col gap-2 w-full lg:w-auto"
      >
        <label
          class="flex items-center gap-2 text-accent-light dark:text-accent-dark font-bold text-lg transition-colors duration-200 hover:text-rick dark:hover:text-morty"
          for="status"
        >
          <svg
            class="text-xl"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M320.2 243.8l-49.7 99.4c-6 12.1-23.4 11.7-28.9-.6l-56.9-126.3-30 71.7H60.6l182.5 186.5c7.1 7.3 18.6 7.3 25.7 0L451.4 288H342.3l-22.1-44.2zM473.7 73.9l-2.4-2.5c-51.5-52.6-135.8-52.6-187.4 0L256 100l-27.9-28.5c-51.5-52.7-135.9-52.7-187.4 0l-2.4 2.4C-10.4 123.7-12.5 203 31 256h102.4l35.9-86.2c5.4-12.9 23.6-13.2 29.4-.4l58.2 129.3 49-97.9c5.9-11.8 22.7-11.8 28.6 0l27.6 55.2H481c43.5-53 41.4-132.3-7.3-182.1z"
            />
          </svg>
           Status
        </label>
        <select
          aria-label="Select status"
          class="px-5 py-3 rounded-xl border-2 border-accent-light/50 dark:border-accent-dark/50 focus:border-rick dark:focus:border-morty focus:ring-4 focus:ring-rick/20 dark:focus:ring-morty/20 bg-white/90 dark:bg-bg-dark/90 text-slate-900 dark:text-white shadow-lg transition-all duration-300 w-full lg:w-40 hover:border-accent-light dark:hover:border-accent-dark hover:shadow-xl hover:scale-[1.02] focus:scale-[1.02] cursor-pointer"
          id="status"
        >
          <option
            class="bg-white dark:bg-bg-dark text-slate-900 dark:text-white"
            value="all"
          >
            All
          </option>
          <option
            class="bg-white dark:bg-bg-dark text-slate-900 dark:text-white"
            value="Alive"
          >
            Alive
          </option>
          <option
            class="bg-white dark:bg-bg-dark text-slate-900 dark:text-white"
            value="Dead"
          >
            Dead
          </option>
          <option
            class="bg-white dark:bg-bg-dark text-slate-900 dark:text-white"
            value="unknown"
          >
            Unknown
          </option>
        </select>
      </div>
      <div
        class="flex flex-col gap-2 w-full lg:w-auto"
      >
        <label
          class="flex items-center gap-2 text-accent-light dark:text-accent-dark font-bold text-lg transition-colors duration-200 hover:text-rick dark:hover:text-morty"
          for="gender"
        >
          <svg
            class="text-xl"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M564 0h-79c-10.7 0-16 12.9-8.5 20.5l16.9 16.9-48.7 48.7C422.5 72.1 396.2 64 368 64c-33.7 0-64.6 11.6-89.2 30.9 14 16.7 25 36 32.1 57.1 14.5-14.8 34.7-24 57.1-24 44.1 0 80 35.9 80 80s-35.9 80-80 80c-22.3 0-42.6-9.2-57.1-24-7.1 21.1-18 40.4-32.1 57.1 24.5 19.4 55.5 30.9 89.2 30.9 79.5 0 144-64.5 144-144 0-28.2-8.1-54.5-22.1-76.7l48.7-48.7 16.9 16.9c2.4 2.4 5.4 3.5 8.4 3.5 6.2 0 12.1-4.8 12.1-12V12c0-6.6-5.4-12-12-12zM144 64C64.5 64 0 128.5 0 208c0 68.5 47.9 125.9 112 140.4V400H76c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h36v36c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12v-36h36c6.6 0 12-5.4 12-12v-40c0-6.6-5.4-12-12-12h-36v-51.6c64.1-14.6 112-71.9 112-140.4 0-79.5-64.5-144-144-144zm0 224c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
            />
          </svg>
           Gender
        </label>
        <select
          aria-label="Select gender"
          class="px-5 py-3 rounded-xl border-2 border-accent-light/50 dark:border-accent-dark/50 focus:border-rick dark:focus:border-morty focus:ring-4 focus:ring-rick/20 dark:focus:ring-morty/20 bg-white/90 dark:bg-bg-dark/90 text-slate-900 dark:text-white shadow-lg transition-all duration-300 w-full lg:w-40 hover:border-accent-light dark:hover:border-accent-dark hover:shadow-xl hover:scale-[1.02] focus:scale-[1.02] cursor-pointer"
          id="gender"
        >
          <option
            class="bg-white dark:bg-bg-dark text-slate-900 dark:text-white"
            value="all"
          >
            All
          </option>
          <option
            class="bg-white dark:bg-bg-dark text-slate-900 dark:text-white"
            value="Male"
          >
            Male
          </option>
          <option
            class="bg-white dark:bg-bg-dark text-slate-900 dark:text-white"
            value="Female"
          >
            Female
          </option>
          <option
            class="bg-white dark:bg-bg-dark text-slate-900 dark:text-white"
            value="Genderless"
          >
            Genderless
          </option>
          <option
            class="bg-white dark:bg-bg-dark text-slate-900 dark:text-white"
            value="unknown"
          >
            Unknown
          </option>
        </select>
      </div>
      <button
        aria-label="Clear filters"
        class="px-4 py-2 rounded bg-cyan-600 text-white font-semibold shadow hover:bg-cyan-700 focus:outline-none focus:ring-2 focus:ring-cyan-400 transition mt-4 lg:mt-8 px-10 py-4 rounded-2xl bg-gradient-to-r from-rick to-morty dark:from-morty dark:to-rick text-white font-bold text-xl shadow-2xl hover:scale-110 hover:shadow-3xl focus:outline-none focus:ring-4 focus:ring-accent-light/50 dark:focus:ring-accent-dark/50 transition-all duration-300 relative overflow-hidden group"
        type="button"
      >
        <svg
          class="inline-block mr-3 text-2xl group-hover:animate-spin transition-transform duration-300"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M370.72 133.28C339.458 104.008 298.888 87.962 255.848 88c-77.458.068-144.328 53.178-162.791 126.85-1.344 5.363-6.122 9.15-11.651 9.15H24.103c-7.498 0-13.194-6.807-11.807-14.176C33.933 94.924 134.813 8 256 8c66.448 0 126.791 26.136 171.315 68.685L463.03 40.97C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.749zM32 296h134.059c21.382 0 32.09 25.851 16.971 40.971l-41.75 41.75c31.262 29.273 71.835 45.319 114.876 45.28 77.418-.07 144.315-53.144 162.787-126.849 1.344-5.363 6.122-9.15 11.651-9.15h57.304c7.498 0 13.194 6.807 11.807 14.176C478.067 417.076 377.187 504 256 504c-66.448 0-126.791-26.136-171.315-68.685L48.97 471.03C33.851 486.149 8 475.441 8 454.059V320c0-13.255 10.745-24 24-24z"
          />
        </svg>
        Clear
      </button>
    </form>
  </section>
</div>
`;

exports[`Snapshot Tests RouteError matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center min-h-[60vh] px-4 text-center animate-fadeIn"
  >
    <img
      alt="404 Portal"
      aria-hidden="true"
      class="w-48 h-48 mb-6 animate-wiggle drop-shadow-lg select-none"
      src="test-file-stub"
    />
    <h1
      class="text-5xl font-extrabold text-cyan-600 dark:text-cyan-400 mb-2"
    >
      404
    </h1>
    <h2
      class="text-2xl font-bold mb-2 text-gray-800 dark:text-gray-100"
    >
      You're Lost in Another Universe!
    </h2>
    <p
      class="mb-6 text-gray-600 dark:text-gray-300 max-w-md"
    >
      The page you're looking for doesn't exist in this universe. Use the portal to return to the home page!
    </p>
    <a
      aria-label="Return to home page"
      class="px-6 py-3 rounded-lg bg-cyan-600 text-white font-semibold shadow hover:bg-cyan-700 focus:outline-none focus:ring-2 focus:ring-cyan-400 transition text-lg animate-fadeIn"
      data-discover="true"
      href="/"
      tabindex="0"
    >
      Return to Home Page
    </a>
  </div>
</div>
`;
